pkg/scheduler/internal/cache/interface.go

```diff
@@ -118,3 +118,11 @@ type Snapshot struct {
 	AssumedPods map[string]bool
 	Nodes       map[string]*schedulernodeinfo.NodeInfo
 }
+
+// NodeInfoSnapshot is a snapshot of cache NodeInfo. The scheduler takes a
+// snapshot at the beginning of each scheduling cycle and uses it for its
+// operations in that cycle.
+type NodeInfoSnapshot struct {
```

Comment:
Is it reasonable for us to have two `Snapshot` types in this package? I don't think I've ever seen what the older one is used for.

Reply:
The older one takes a snapshot of the cache for the "cache comparer". It is invoked when SIGUSR2 is sent to the scheduler.